scripts/AbstractWidget.py

```diff
@@ -0,0 +1,322 @@
+import re
+import sys
+import os
+class eventHandlerInfo:
+    def __init__(self, java_src, identity, java_line_number,xml_line_number = None, xml_src = None, name=None):
+        self.xml_src = xml_src
+        self.java_src = java_src
+        self.identity = identity
+        self.java_line_number = java_line_number
+        self.xml_line_number = xml_line_number
+        self.name = name
+
+    def toString(self):
+        result = []
+        result.append("Java Source: " + self.java_src + ":" + self.java_line_number)
+        if self.xml_src != None:
+            result.append("XML source:" + self.xml_src + ":" + str(self.xml_line_number))
+        return result
+
+    def getName(self):
+        return self.name
+
+class dialogUI:
+    def __init__(self, node):
```

Comment:
What is the type of `node`? (You can add type signatures in Python as well http://mypy-lang.org)